### PR TITLE
Notebooks page for custom docker image 

### PR DESCRIPTION
**Pull request by samumantha**
Adds detailed instructions for custom docker image creation and usage in Notebooks.
Some of this is now duplicate from 'getting started' page. The thought behind it is that personally I rather have all information needed on one page rather than jumping back and forth between pages.

I would have some follow up suggestions on how to remove having duplicates and also incorporate @ktiits suggestion on splitting instructions between course participant and course creator, but that is stuff for a future PR coming soon. :)

What do you think, @tourunen, @Ilavarasi1? Is this any help here?

### CODE REVIEW

**Comment by samumantha** [comment]
To do:
- [ ] add example minimal dockerfile for Rstudio and Jupyter with pip (tabs?)
- [ ] fix conda base bug and update dockerfile example accordingly
- [ ] reduce to only custom image part, without all steps that are already in 'getting started'

**Comment by samumantha** [comment]
Adding packages/ creating new environemt with environment.yml ?
- freeze install?

**Comment by samumantha** [comment]
back to WIP: talked to Notebooks team, will update this PR with more suggestions on reorganizing notebooks docs :  participants vs course organizers and merging "using provided notebook image" and "using custom image"

**Comment by samumantha** [comment]
As discussed in ToolDS meeting and Rocketchat, the Notebooks page is now restructured. The main change is that things are split in instructions for students and teachers.
What do you think? 
Anything missing?  

One thing I am not happy with is that the "Notebooks for collaboration" is now under "guide for teachers" , any better idea?

**Comment by ktiits** [comment]
In my opinion having collaboration in teacher's guide is fine. It otherwise would just be a duplicate again. Rewrote a little the students guide, so that is one (with option to use join code) rather than 2 separate topics. Did also minor changes to teachers guide, mostly about style (italic away).

Students and teachers guides now fine by me.

I still find the Concepts (and Data persistance) page not nice for users, I would rather suggest to have proper intro to introduce the service, may-be with some image to illustrate the concepts? Such definitions page is very difficult to follow and understand. But I would merge this PR as such and then discuss the rest..

File: guide_for_teachers.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+   # copy the desired installation script into docker file system, make sure that you have execute rights to the script
+   COPY install_xx.sh /rocker_scripts/
+
+   # install the custom packages and system dependencies by running the script
+   RUN /rocker_scripts/install_xx.sh
+
+   # Rtsudio is exposed on port 8787
+   EXPOSE 8787
+
+   CMD ["/init"]
```
Comment (samumantha) [inline guide_for_teachers.md:100]: Is this needed?

File: docs/cloud/csc_notebooks/guide_for_teachers.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    # it has eg conda already installed
+    FROM jupyter/minimal-notebook
+
+    # add your name as maintainer, with your email address for future questions
+    LABEL maintainer="your-name-here"
+
+    #some first setup steps need to be run as root user
+    USER root
+
+    # set home environment variable to point to user directory
```
Comment (samumantha) [inline docs/cloud/csc_notebooks/guide_for_teachers.md:50]:
```suggestion
    # following line sets the home environment variable to point to user directory
```

File: docs/cloud/csc_notebooks/guide_for_teachers.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    ENV HOME /home/$NB_USER
+
+    # install needed extra tools, eg ssh-client and less
+    RUN apt-get update \
+        && apt-get install -y ssh-client less \
+        && apt-get clean
+
+    # the user set here will be the user that students will use
```
Comment (samumantha) [inline docs/cloud/csc_notebooks/guide_for_teachers.md:58]:
```suggestion
    # below line sets the user to NB_USER, i.e. unsets root 
```

**Review by ktiits — APPROVED** [review]

**Pull request merged**